lib/src/metta/types.rs

```diff
@@ -631,4 +640,157 @@ mod tests {
         assert!(check_type(&space, &atom("(Refl TermGnd)"), &atom("(=== $a $a)")));
         assert!(check_type(&space, &atom("(Refl TermGnd)"), &atom("(=== $a $b)")));
     }
+
+    #[test]
+    fn check_type_accept_meta_type() {
+        let type_r = &atom("R");
+        let space = metta_space("
+            (: R Type)
+            (: A Type)
+            (: B Type)
+            (: a A)
+            (: b B)
+
+            (: aF (-> A R))
+            (: atomF (-> Atom R))
+            (: exprF (-> Expression R))
+            (: gndF (-> Grounded R))
+            (: symF (-> Symbol R))
+            (: varF (-> Variable R))
+        ");
+
+        assert!(check_type(&space, &atom("(aF a)"), type_r));
+        assert!(!check_type(&space, &atom("(aF b)"), type_r));
+        // TODO: (aF b) is incorrectly typed, but it is an Atom and check_type
```

Comment:
@Necr0x0Der please look at this and below examples marked by TODO. Should we handle them differently?

Reply:
We need some real use cases in order to understand if this is the intended behavior or not. Let us keep it as is for now.